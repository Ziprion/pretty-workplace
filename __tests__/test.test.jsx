import React from 'react';

import { LoginPage } from '@pages';
import { render, renderSnapshot, screen } from '@test-utils';

import '@testing-library/jest-dom';

test('test', () => {
  const tree = renderSnapshot(<LoginPage />);

  expect(tree).toMatchSnapshot();
});

test('test2', () => {
  const tree = renderSnapshot(<LoginPage />, [ '/signup' ]);

  expect(tree).toMatchSnapshot();
});

test('test3', async () => {
  render(<LoginPage />, [ '/signup' ]);

  expect(screen.getByRole('heading')).toHaveTextContent('pretty workplace');
});
