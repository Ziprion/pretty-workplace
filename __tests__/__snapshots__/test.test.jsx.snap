// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test 1`] = `
.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  min-width: 100%;
  height: 34px;
  padding: 6px 16px;
  color: #ffffff;
  font-size: 14px;
  line-height: 21px;
  background: #0078d4;
  border: 1px solid #0078d4;
  border-radius: 4px;
  cursor: pointer;
  -webkit-transition: 0.1s ease;
  transition: 0.1s ease;
}

.c12 svg {
  width: 100%;
  height: 100%;
}

.c12:hover,
.c12:focus {
  color: #ffffff;
  background: #005ba1;
  border: 1px solid #005ba1;
}

.c12:disabled {
  color: #37415142;
  background: #e6e7e9;
  border: 1px solid #e6e7e9;
  cursor: not-allowed;
}

.c9 {
  width: 100%;
  min-width: 250px;
  padding: 8px 16px;
  color: #111827;
  background: #ffffff;
  border: 1px solid #D1D5DB;
  border-radius: 4px;
  -webkit-user-select: auto;
  -moz-user-select: auto;
  -ms-user-select: auto;
  user-select: auto;
}

.c9::-webkit-input-placeholder {
  color: #6B7280;
}

.c9::-moz-placeholder {
  color: #6B7280;
}

.c9:-ms-input-placeholder {
  color: #6B7280;
}

.c9::placeholder {
  color: #6B7280;
}

.c9:focus {
  outline: 1px solid #6B7280;
}

.c9:disabled {
  background: #e6e7e9;
  border: 1px solid #e6e7e9;
  cursor: not-allowed;
}

.c9:disabled::-webkit-input-placeholder {
  color: #37415142;
  opacity: 1;
}

.c9:disabled::-moz-placeholder {
  color: #37415142;
  opacity: 1;
}

.c9:disabled:-ms-input-placeholder {
  color: #37415142;
  opacity: 1;
}

.c9:disabled::placeholder {
  color: #37415142;
  opacity: 1;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c3 {
  width: 24px;
  height: 24px;
  margin-right: 8px;
}

.c4 {
  color: #111827;
  font-size: 18px;
  line-height: 25px;
  text-transform: uppercase;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c8 {
  display: block;
  margin: 8px 0;
}

.c10 {
  height: 18px;
  color: #dc3545;
  font-size: 12px;
  line-height: 18px;
}

.c7 {
  position: relative;
  width: 100%;
  text-align: center;
}

.c13 {
  min-width: 80px;
  margin-left: 8px;
}

.c11 {
  height: 16px;
}

.c14 {
  margin-top: 32px;
  font-size: 12px;
  line-height: 18px;
}

.c15 {
  color: #0078d4;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c15:hover {
  color: #005ba1;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  min-height: 100vh;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  max-width: 400px;
  padding: 40px 32px 24px;
  background: #ffffff;
  border-radius: 4px;
  box-shadow: 0 10px 15px -3px #64748b1f, 0 4px 6px -2px #64748b0d;
}

.c5 {
  margin: 16px 0;
  color: #6B7280;
  font-size: 12px;
  line-height: 18px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <img
        alt="pretty workplace"
        className="c3"
        size="small"
        src="images/icons/logo-small.png"
      />
      <h1
        className="c4"
      >
        pretty workplace
      </h1>
    </div>
    <span
      className="c5"
    >
      Sign in to continue
    </span>
    <form
      className="c6"
      onSubmit={[Function]}
    >
      <div
        className="c7"
      >
        <label
          className="c8"
          htmlFor="email"
        >
          Email
        </label>
        <input
          autoFocus={true}
          className="c9"
          disabled={false}
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Enter your email"
          type="text"
          value=""
          width="100%"
        />
        <div
          className="c10"
        >
          
        </div>
      </div>
      <div
        className="c7"
      >
        <label
          className="c8"
          htmlFor="password"
        >
          Password
        </label>
        <input
          autoFocus={false}
          className="c9"
          disabled={false}
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Enter your password"
          type="password"
          value=""
          width="100%"
        />
        <div
          className="c10"
        >
          
        </div>
      </div>
      <div
        className="c10"
      >
        
      </div>
      <div
        className="c11"
      />
      <button
        className="c12 c13"
        disabled={false}
        type="submit"
        width="100%"
      >
        Sign in
      </button>
      <div
        className="c14"
      >
        Do not have an account? 
        <a
          className="c15"
          href="/signup"
          onClick={[Function]}
        >
          Create an account
        </a>
      </div>
    </form>
  </div>
</div>
`;

exports[`test2 1`] = `
.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  min-width: 100%;
  height: 34px;
  padding: 6px 16px;
  color: #ffffff;
  font-size: 14px;
  line-height: 21px;
  background: #0078d4;
  border: 1px solid #0078d4;
  border-radius: 4px;
  cursor: pointer;
  -webkit-transition: 0.1s ease;
  transition: 0.1s ease;
}

.c12 svg {
  width: 100%;
  height: 100%;
}

.c12:hover,
.c12:focus {
  color: #ffffff;
  background: #005ba1;
  border: 1px solid #005ba1;
}

.c12:disabled {
  color: #37415142;
  background: #e6e7e9;
  border: 1px solid #e6e7e9;
  cursor: not-allowed;
}

.c9 {
  width: 100%;
  min-width: 250px;
  padding: 8px 16px;
  color: #111827;
  background: #ffffff;
  border: 1px solid #D1D5DB;
  border-radius: 4px;
  -webkit-user-select: auto;
  -moz-user-select: auto;
  -ms-user-select: auto;
  user-select: auto;
}

.c9::-webkit-input-placeholder {
  color: #6B7280;
}

.c9::-moz-placeholder {
  color: #6B7280;
}

.c9:-ms-input-placeholder {
  color: #6B7280;
}

.c9::placeholder {
  color: #6B7280;
}

.c9:focus {
  outline: 1px solid #6B7280;
}

.c9:disabled {
  background: #e6e7e9;
  border: 1px solid #e6e7e9;
  cursor: not-allowed;
}

.c9:disabled::-webkit-input-placeholder {
  color: #37415142;
  opacity: 1;
}

.c9:disabled::-moz-placeholder {
  color: #37415142;
  opacity: 1;
}

.c9:disabled:-ms-input-placeholder {
  color: #37415142;
  opacity: 1;
}

.c9:disabled::placeholder {
  color: #37415142;
  opacity: 1;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c3 {
  width: 24px;
  height: 24px;
  margin-right: 8px;
}

.c4 {
  color: #111827;
  font-size: 18px;
  line-height: 25px;
  text-transform: uppercase;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c8 {
  display: block;
  margin: 8px 0;
}

.c10 {
  height: 18px;
  color: #dc3545;
  font-size: 12px;
  line-height: 18px;
}

.c7 {
  position: relative;
  width: 100%;
  text-align: center;
}

.c13 {
  min-width: 80px;
  margin-left: 8px;
}

.c11 {
  height: 16px;
}

.c14 {
  margin-top: 32px;
  font-size: 12px;
  line-height: 18px;
}

.c15 {
  color: #0078d4;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c15:hover {
  color: #005ba1;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  min-height: 100vh;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  max-width: 400px;
  padding: 40px 32px 24px;
  background: #ffffff;
  border-radius: 4px;
  box-shadow: 0 10px 15px -3px #64748b1f, 0 4px 6px -2px #64748b0d;
}

.c5 {
  margin: 16px 0;
  color: #6B7280;
  font-size: 12px;
  line-height: 18px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <img
        alt="pretty workplace"
        className="c3"
        size="small"
        src="images/icons/logo-small.png"
      />
      <h1
        className="c4"
      >
        pretty workplace
      </h1>
    </div>
    <span
      className="c5"
    >
      Sign up to continue
    </span>
    <form
      className="c6"
      onSubmit={[Function]}
    >
      <div
        className="c7"
      >
        <label
          className="c8"
          htmlFor="email"
        >
          Email
        </label>
        <input
          autoFocus={true}
          className="c9"
          disabled={false}
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Enter your email"
          type="text"
          value=""
          width="100%"
        />
        <div
          className="c10"
        >
          
        </div>
      </div>
      <div
        className="c7"
      >
        <label
          className="c8"
          htmlFor="firstName"
        >
          Name
        </label>
        <input
          autoFocus={false}
          className="c9"
          disabled={false}
          id="firstName"
          name="firstName"
          onChange={[Function]}
          placeholder="Enter your name"
          type="text"
          value=""
          width="100%"
        />
        <div
          className="c10"
        >
          
        </div>
      </div>
      <div
        className="c7"
      >
        <label
          className="c8"
          htmlFor="lastName"
        >
          Surname
        </label>
        <input
          autoFocus={false}
          className="c9"
          disabled={false}
          id="lastName"
          name="lastName"
          onChange={[Function]}
          placeholder="Enter your surname"
          type="text"
          value=""
          width="100%"
        />
        <div
          className="c10"
        >
          
        </div>
      </div>
      <div
        className="c7"
      >
        <label
          className="c8"
          htmlFor="password"
        >
          Password
        </label>
        <input
          autoFocus={false}
          className="c9"
          disabled={false}
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Enter your password"
          type="password"
          value=""
          width="100%"
        />
        <div
          className="c10"
        >
          
        </div>
      </div>
      <div
        className="c7"
      >
        <label
          className="c8"
          htmlFor="confirm"
        >
          Confirm password
        </label>
        <input
          autoFocus={false}
          className="c9"
          disabled={false}
          id="confirm"
          name="confirm"
          onChange={[Function]}
          placeholder="Enter your password again"
          type="password"
          value=""
          width="100%"
        />
        <div
          className="c10"
        >
          
        </div>
      </div>
      <div
        className="c10"
      >
        
      </div>
      <div
        className="c11"
      />
      <button
        className="c12 c13"
        disabled={false}
        type="submit"
        width="100%"
      >
        Sign up
      </button>
      <div
        className="c14"
      >
        Already have an account? 
        <a
          className="c15"
          href="/signin"
          onClick={[Function]}
        >
          Sign in
        </a>
      </div>
    </form>
  </div>
</div>
`;
